vendor/github.com/moby/vpnkit/go/pkg/vmnet/vmnet.go

```diff
@@ -563,7 +563,9 @@ func (v *Vif) dhcp() (net.IP, error) {
 	ethernet := NewEthernetFrame(broadcastMAC, v.ClientMAC, 0x800)
 	ethernet.setData(ipv4.Bytes())
 
-	file, err := os.Create("/tmp/go.pcap")
+	runDir := os.Getenv("XDG_RUNTIME_DIR")
```

Comment:
not guaranteed to be set

Reply:
If set, this variable is set (for example) for user 1001 to /var/run/1001, which is properly owned by uid 1001.
This is true (at least when using rootlesskit to launch dockerd) whether running both outside the rootless context (when user.Uid is 1000) and inside (when user.Uid is 0).

Can you suggest another way to get real users' correct subdirectory of /run/user/ ?

Reply:
One alternative is to fall back to the old behavior when the `XDG_RUNTIME_DIR` is not set.  

I am not a go programmer, but I have gotten the impression that, if the environment variable is not set, `runDir` will be set to an empty string (and I could test for that condition) - is that correct?